src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -782,9 +816,6 @@ void checkQueryOptionsSupportability(QueryOptions queryOptions) {
                 && !queryOptions.getLockOptions().isEmpty()) {
             throw new FeatureNotSupportedException("'lockOptions' in QueryOptions not supported");
         }
-        if (queryOptions.getComment() != null) {
-            throw new FeatureNotSupportedException("TO-DO-HIBERNATE-53 https://jira.mongodb.org/browse/HIBERNATE-53");
-        }
```

Comment:
Why was this check removed?

Reply:
because from this PR, the `queryOptions.getComment()` would be set to the HQL string, so the above statement  had to be removed.

Reply:
if comment is explicitly set as below, the `queryOptions.getComment()` will be the set accordingly:
```
            var results = session.createSelectionQuery(
                            "select name, age from Contact where country = :country", Object[].class)
                    .setParameter("country", Country.CANADA.name())
                    .setComment("customized comment here")
                    .getResultList();
```
so the ticket is still valid, but for HQL based selection query, the comment will be set to HQL string by default but won't be null any more (unless `setComment(null)` is used above).

Reply:
That's quite unfortunate: we don't support `queryOptions.getComment()`, yet we can't really inform an application that we ignore it if an application specifies a comment because Hibernate ORM also specifies it implicitly. This situation is similar to `TableMutation.getMutationComment`, which Hibernate ORM specifies on its own.

Reply:
It may be useful for speeding up reviews to explicitly point reviewers to such changes in advance, with an explanation behind the change, given that it's not obvious.

Reply:
yeah, good advice.